### PR TITLE
[k8s-extension] Update extension CLI to v1.4.3

### PR DESCRIPTION
**Pull request by bavneetsingh16**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-extension

File: src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def _is_downgrade(v1: str, v2: str) -> bool:
         Returns True if version v1 is less than version v2.
         """
         try:
-            return packaging_version.Version(v1) < packaging_version.Version(v2)
-        except packaging_version.InvalidVersion:
+            return VersionInfo.parse(v1) < VersionInfo.parse(v2)
+        except ValueError:
```
Comment (zhoxing-ms) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py:208]: May I ask why do you need to use `semver` instead of `packaging`?

Comment (bavneetsingh16) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py:208]: The reason is mentioned in [https://github.com/AzureArcForKubernetes/azure-cli-extensions/pull/241](https://github.com/AzureArcForKubernetes/azure-cli-extensions/pull/241). Please let me know if you need more information.

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
Approving for the usage of `semver` part. 

BTW, there are 85 commits in this PR. It's better to rebase it to `main` and clean up the history.

**Comment by zhoxing-ms** [comment]
> BTW, there are 85 commits in this PR. It's better to rebase it to main and clean up the history.

@bavneetsingh16 Please help rebase the code

**Pull request closed without merging**